packages/sns/lib/utils/stsUtils.ts

```diff
@@ -0,0 +1,14 @@
+import { GetCallerIdentityCommand, type STSClient } from '@aws-sdk/client-sts'
+
+/**
+ * Manually builds the ARN of a topic based on the current AWS account and the topic name.
+ * It follows the following pattern: arn:aws:sns:<region>:<account-id>:<topic-name>
+ * Doc -> https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html
+ */
+export const buildTopicArn = async (client: STSClient, topicName: string) => {
+  const identityResponse = await client.send(new GetCallerIdentityCommand({}))
```

Comment:
I would move this call outside of the function and make building itself synchronous, but also expose an asynchronous method on top of it that does resolution for you.

from performance perspective it would be ideal to only resolve the identity once per app startup and not once per every queue. for now we can rely on async one, but having sync one exposed as well would allow to optimize in the future by moving identity resolution to app itself

Reply:
I am not sure if moving identity resolution is a good idea 🤔, maybe we can do it once and cache the result so the next calls won't need it, we can even do it within this file, what do you think?

Reply:
sounds good! but let's expose a method to reset it as well, for test e. g.